src/context/NotesProvider.jsx

```diff
@@ -0,0 +1,63 @@
+import { useReducer, useContext, createContext } from 'react';
+
+const initialNotes = [
+  { id: Date.now(), text: 'Get cold brew please!', done: false },
+];
+// an array of info
+const noteReducer = (state, action) => {
+  switch (action.type) {
+    case 'add_note':
+      return [
+        { id: Date.now(), text: action.payload.text, done: false },
+        ...state,
+      ];
+    case 'update_note':
+      return state.map((note) => {
+        if (note.id === action.payload.note.id) {
+          const { done, text } = action.payload.note;
+          return {
+            ...note,
+            done,
+            text,
+          };
+        }
+        return note;
+      });
+    case 'delete_note':
+      return notes.filter((note) => note.id !== action.payload.id);
+
```

Comment:
It appears notes is not defined, consider filtering through state instead of notes here.